fhir-server-test/src/test/java/com/ibm/fhir/server/test/SearchReverseChainTest.java

```diff
@@ -0,0 +1,1034 @@
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON)
+                .get();
+        assertResponse(response, Response.Status.BAD_REQUEST.getStatusCode());
+        assertExceptionOperationOutcome(response.readEntity(OperationOutcome.class),
+                "Undefined Modifier: badModifier");
+    }
+
+    @Test(groups = { "server-search-reverse-chain" })
+    public void testSearchReverseChainWithSearchParmUnsupportedModifierError() {
+        WebTarget target = getWebTarget();
+        Response response =
+                target.path("Patient")
+                .queryParam("_has:Procedure:subject:code:contains", "test")
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON)
+                .get();
+        assertResponse(response, Response.Status.BAD_REQUEST.getStatusCode());
+        assertExceptionOperationOutcome(response.readEntity(OperationOutcome.class),
+                "Unsupported type/modifier combination: token/contains");
+    }
+
+    @Test(groups = { "server-search-reverse-chain" })
+    public void testSearchReverseChainWithSearchParmIsResultParmError() {
+        WebTarget target = getWebTarget();
+        Response response =
+                target.path("Patient")
+                .queryParam("_has:Procedure:subject:_total", "1")
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON)
+                .get();
+        assertResponse(response, Response.Status.BAD_REQUEST.getStatusCode());
+        assertExceptionOperationOutcome(response.readEntity(OperationOutcome.class),
+                "Search parameter '_total' for resource type 'Procedure' was not found.");
+    }
+
+    @Test(groups = { "server-search-reverse-chain" })
+    public void testSearchReverseChainWithChainedSearchParmBadModifierError() {
+        WebTarget target = getWebTarget();
+        Response response =
+                target.path("Patient")
+                .queryParam("_has:Procedure:subject:encounter:contains.status", "test")
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON)
+                .get();
+        assertResponse(response, Response.Status.BAD_REQUEST.getStatusCode());
+        assertExceptionOperationOutcome(response.readEntity(OperationOutcome.class),
+                "Modifier: 'CONTAINS' not allowed on chained parameter");
+    }
+
+    @Test(groups = { "server-search-reverse-chain" })
+    public void testSearchReverseChainWithChainedSearchParmNotReferenceError() {
+        WebTarget target = getWebTarget();
+        Response response =
+                target.path("Patient")
+                .queryParam("_has:Procedure:subject:code.status", "test")
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON)
+                .get();
+        assertResponse(response, Response.Status.BAD_REQUEST.getStatusCode());
+        assertExceptionOperationOutcome(response.readEntity(OperationOutcome.class),
+                "Type: 'TOKEN' not allowed on chained parameter");
```

Comment:
Why does this become uppercase? I think it's from the fhir-search Type enum, and we should probably use the natural value of lower case token which is what's read fromt he Parameter

Reply:
changed the code in SearchUtil that constructs the exception to use Token.value() so we get the natural value